src/targets/gpu/include/migraphx/gpu/mlir_offload.hpp

```diff
@@ -35,15 +35,15 @@ namespace gpu {
 
 MIGRAPHX_GPU_EXPORT bool mlir_enabled();
 
-struct MIGRAPHX_GPU_EXPORT fuse_mlir
+struct MIGRAPHX_GPU_EXPORT mlir_offload
```

Comment:
The class should be named `fuse_mlir`.

Reply:
But then it sounds as if we only provide fused operation. However, this PR aims at additionally providing standalone convs.

https://github.com/ravil-mobile/AMDMIGraphX/blob/81aa21794803cdd73e5c88dfb7bf459914b74d8a/src/targets/gpu/fuse_mlir.cpp#L335-L349

Moreover, we are going to add standalone GEMMs. Additionally, we would like to have an option to switch off fused ops based on the value of an env. variables. In this case, we would like to completely fall back to standalone GEMMs and CONVs taking them from rocMLIR; we think it may be help us during our performance analyses and monitoring.

By and large, the `apply` method is not supposed to be focused on fused operations only; it is supposed to be generic. That is the reason why @krzysz00 and I proposed to rename the struct `mlir_offload`.

Reply:
> But then it sounds as if we only provide fused operation.

For fuse_ck, we handle standalone gemms, for fuse_pointwise we handle standalone pointwise, and fuse_reduce we handle standalone reduce. So I dont see how it would be different here.

> By and large, the apply method is not supposed to be focused on fused operations only; it is supposed to be generic.

Yea niether do fuse_ck, fuse_pointwise, and fuse_reduce focus on fused operations only. 

> That is the reason why @krzysz00 and I proposed to rename the struct mlir_offload.

Yes, but such renaming should go in a separate PR, because the other fuse passes need to renamed as well.

Reply:
> Yes, but such renaming should go in a separate PR, because the other fuse passes need to renamed as well.

Got your point! Ok, I will rework.

Reply:
done